### ISSUE TITLE
Turning on test coverage rebuilds all packages

### ISSUE THREAD

**Issue by hesselink** [commercialhaskell/stack#1940]
If I call `stack test --coverage <some-package>` in a repository with many packages, all packages (or at least dependencies of "some-package") will be reinstalled with a coverage flag passed to it. This causes a lot of rebuilding (46 packages in my case) and probably generate coverage reports for all of them. In my case, I'm only interested of the coverage of the package I'm testing. Is there a way to tell stack this? If not, would it be possible to add?

**Comment by mgsloan** [comment]
Good point! I've opened #2417 to track a potential general resolution to this.  Marking this as blocked on that.

**Comment by chreekat** [comment]
Related: 'stack test --coverage' will rebuild (and make coverage reports for) packages specified as extra-deps via 'package [{location: ..., extra-dep: true}]`

**Comment by rlpowell** [comment]
This is quite onerous for me because one of my dependencies is an "extra-dep: true" local copy of pandoc -___- because I rely on some things in pandoc mainline.  Pandoc takes a loooooong time to compile; easily 10x the length of time it takes to compile the package that I actually need to compile.

I'm not an expert at stack itself or Cabal, but if someone points me in the right direction I'd love to fix this.

**Comment by rlpowell** [comment]
What's worse is that they all rebuild *every time*, even though nothing has changed in the subordinate packages.  So a one-line change to my code requires a complete build of my local copy of pandoc if I want a new coverage report.  I'm not sure that was clear in the original issue statement.

**Comment by chreekat** [comment]
@rlpowell tell me about it. :)

Have you tried the newest version of stack? It changed how "special" extra-deps are specified. I haven't changed it yet, but I am optimistic that it fixes this set of related problems.

**Comment by rlpowell** [comment]
Despite documentation changes in master, stack 1.5.0 still does the "packages: extra-dep: true" thing; I couldn't get local directories to work under "extra-deps:", which I think is the thing you're pointing at?  If that's supposed to work in 1.5.0, I should file a bug.  If that's not the thing you are pointing at, please explain.  If you mean "have you tried master-branch latest stack?", then no.

**Comment by chreekat** [comment]
I was mistaken about released versions. The changes I'm talking about
should drop in 1.6, I think.

Writeup:
https://www.fpcomplete.com/blog/2017/07/stacks-new-extensible-snapshots
(Unreleased) docs:
https://docs.haskellstack.org/en/latest/yaml_configuration/#packages-and-extra-deps

**Comment by rlpowell** [comment]
Welllll, I can confirm that in current mainline, this problem is definitely "solved", with an emphasis on the scare quotes :D.  See #3306
